### ISSUE TITLE
Using autounattend.xml for initializing desktops or laptops with preinstalled Windows

### ISSUE THREAD

**Issue by bfg01** [memstechtips/UnattendedWinstall#127]
I first read the Microsoft docs about answer files before getting here, in order to get the basic idea first.

As far as I understood, there are 2 types of answer files: unattend.xml for only the Windows PE part, this is, the "Windows installer"; and the autounattend.xml for all what follows, which is the OOBE steps (keyboard layouts, connecting to WiFi, setting Microsoft account or local accounts, etc).

Having read here, it seems a single autounattend.xml could cover both phases of the installation in one single file...

Assuming this is true, can one also use autounattend.xml from this project to "initialize" computers with preinstalled Windows? Which would be only the OOBE steps without the initial installation steps.
If positive, how would I load the xml here?
Or would using this actually mess up the preinstalled partitions?

Thanks.

**Comment by memstechtips** [comment]
A lot of people have asked about this, but I have not been able to apply a `autounattend.xml` file on an existing Windows installation or upgrade.

**Comment by bfg01** [comment]
I found a way in another forum, albeit a very "partial" way...
Let's assume using the "Standard" one from this repo.

--Put the xml in the root of any USB drive, and rename it to "unattend.xml"
--Once laptop is powered on and OOBE (Windows welcome) just starts, plug USB drive, hit Shift+F10, and type:
%WINDIR%\System32\Sysprep\sysprep.exe /oobe /reboot /unattend:E:\unattend.xml
(Change E: to the actual letter in use)

System reboots and begins processing just the "oobeSystem" pass.
Of course, this implies **nothing** from "Specialize" pass is processed, and seemingly neither anything before this pass.
I think, this is because computers with preinstalled Windows precisely ship with preinstalled passes up to the OOBE one, this being the interactive part.

But since almost the entire content of the xml is in the "Specialize" pass, I wonder how useful it would be, aside saving us all the keyboard typing and clicks from powering on to first log in...

**Comment by havr-p** [comment]
@bfg01 hello, currently I am solving similiar task. Was it your final decision?

**Comment by bfg01** [comment]
Once I get a chance of initializing a Windows box again I'll post here again.

**Comment by havr-p** [comment]
thank you!

**Comment by bfg01** [comment]
@memstechtips 
Just today I was asked to initialize a laptop with Windows preinstalled and I think I was finally able to complete what was started in this issue thread.
So I think just until now the issue would be really closed, in case this is useful for you and other users.

What I posted in post number 3 here was just half true; it is totally possible to initialize preinstalled Windows with autounattend.xml, no missing passes.
The thing is, there was something missing in the sysprep command, and the complete one would be:
%WINDIR%\System32\Sysprep\sysprep.exe /generalize /oobe /reboot /unattend:E:\unattend_name.xml
I think this processes all passes from the "Generalize" one onward, and the "Specialize" pass goes after the "Generalize" one, so it's included and the functionality is the expected one.
Also, I put "unattend_name.xml" to emphasize the fact that one can use whatever name wanted for the XML, as long as using with sysprep command.

I also tested a second method, which would probably look more like a "trick" --not sure:
--Put the XML in the root of any USB drive, and rename it to "unattend.xml"
--In the XML, within the component "Microsoft-Windows-Deployment", order number 1 which is the PowerShell command creating the Extensions script for uninstalling apps, change the part saying 'C:\Windows\Panther\unattend.xml' to 'E:\unattend.xml'
--Plug USB drive **before** powering on the PC for the first time

With this the initialization starts automatically once the PC is powered on.
But there are caveats:
--Plug USB drive **before** powering on the PC for the first time. Else even if rebooting it will just re-enter directly to the OOBE and the only way left is using sysprep command.
--For the example I used drive letter E, but it totally depends on what letter the USB drive is assigned. Probably a modification in the XML with a script scanning all USB drives with "unattend.xml" would be needed for this...

Well, these were my final findings. Hope they serve for others.
